### PR TITLE
plugins/refactoring: init + test

### PR DESCRIPTION
**Pull request by hmajid2301**
Add new plugin [refactoring.nvim])https://github.com/ThePrimeagen/refactoring.nvim)

### CODE REVIEW

File: plugins/utils/refactoring.nix [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+      package = helpers.mkPackageOption "refactoring.nvim" pkgs.vimPlugins.refactoring-nvim;
+
+      promptFuncReturnType =
+        helpers.defaultNullOpts.mkNullable
+        (
+          with types;
+            attrsOf bool
+        )
+        ''
+          {
```
Comment (hmajid2301) [inline plugins/utils/refactoring.nix:24]: Im not sure if this it the best approach ?

File: plugins/utils/refactoring.nix [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+  config = let
+    cfg = config.plugins.refactoring;
+  in
+    mkIf cfg.enable {
+      extraPlugins = [cfg.package];
+
+      extraConfigLua = let
+        opts = {
+          inherit (cfg);
+          prompt_func_return_type = promptFuncReturnType;
```
Comment (hmajid2301) [inline plugins/utils/refactoring.nix:155]: do we need to do any transformation here ?

File: plugins/utils/refactoring.nix (hunk 1 above)
Comment (traxys) [reply plugins/utils/refactoring.nix:24]: I think it is

File: plugins/utils/refactoring.nix [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+        )
+        ''
+          {
+            "go" = "false";
+            "java" = "false";
+            "cpp" = "false";
+            "c" = "false";
+            "h" = "false";
+            "hpp" = "false";
+            "cxx" = "false";
```
Comment (traxys) [inline plugins/utils/refactoring.nix:31]: There are no quotes around "false"

File: plugins/utils/refactoring.nix [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
```
Comment (traxys) [inline plugins/utils/refactoring.nix:54]: No quotes around false

File: plugins/utils/refactoring.nix (hunk 2 above)
Comment (traxys) [reply plugins/utils/refactoring.nix:155]: I don't think so

**Review by traxys — CHANGES_REQUESTED** [review]

File: plugins/utils/refactoring.nix (hunk 2 above)
Comment (traxys) [reply plugins/utils/refactoring.nix:155]: You need to do `XXXX = cfg.XXXX;` though (see the CI fail)

Comment (hmajid2301) [reply plugins/utils/refactoring.nix:154]: I had it committed, just not pushed!

File: plugins/utils/refactoring.nix (hunk 4 above)
Comment (hmajid2301) [reply plugins/utils/refactoring.nix]: done

File: plugins/utils/refactoring.nix [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+          For certain languages like Golang, types are required for functions that return an object(s) and parameters of functions.
+          Unfortunately, for some parameters and functions there is no way to automatically find their type. In those instances,
+          we want to provide a way to input a type instead of inserting a placeholder value.
+        '';
+
+      printVarStatements =
+        helpers.defaultNullOpts.mkNullable
+        (
+          with types;
+            attrOf (listOf str)
```
Comment (traxys) [inline plugins/utils/refactoring.nix:67]: CI fails due to typo: attrOf --> attrsOf
You should run `nix flake check`, it would find those sorts of things

**Review by traxys — CHANGES_REQUESTED** [review]

File: plugins/utils/refactoring.nix (hunk 5 above)
Comment (hmajid2301) [reply plugins/utils/refactoring.nix:67]: Yeh sorry, I just usually like to push my changes. I'll try to check before pushing :sweat:

**Review by traxys — APPROVED** [review]

**Pull request merged**